File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    if n == 1:
        return 1
    
    d = [0] * (n + 1)
    d[1] = 1
    d[2] = 2

    for i in range(3, n + 1):
        d[i] = (d[i - 1] + d[i - 2]) % 1_234_567
    
    return d[n]

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(long n) {
        if (n == 1) return 1;

        long[] d = new long[(int) n + 1];
        d[1] = 1;
        d[2] = 2;

        for (int i = 3; i <= n; i++) {
            d[i] = (d[i - 1] + d[i - 2]) % 1234567;
        }

        return d[(int) n];
    }
}